### PR TITLE
Added polling for the SingleFileMetastore.

### LINKED ISSUE #839: Search servers should (optionally?) poll the S3 metastore for updates
Right now, if we have 
- one indexer
- search servers
and we are using the S3 metastore, the search server will load the S3 metastore state on startup and never update it.

We could have an option to reload it at regular interval
(once every 10s induce a cost of $0.1 per month per index).

### PR DESCRIPTION
**Pull request by fulmicoton**
This adds the possibility to configure a polling period for a given SingleFileMetastore by appending a uri fragment to tis uri like this: `s3://my_bucket/blabla#period=30s`

Closes #839

### CODE REVIEW

File: quickwit-metastore/src/metastore/single_file_metastore/single_file_metastore_factory.rs [hunk 1]
```diff
@@ -40,27 +43,81 @@ impl Default for SingleFileMetastoreFactory {
     }
 }
 
+fn extract_polling_period_from_uri(uri: &str) -> (String, Option<Duration>) {
```
Comment (guilload) [inline quickwit-metastore/src/metastore/single_file_metastore/single_file_metastore_factory.rs:46]: I'm not fond of this. Unfortunately, this adds another way to configure quickwit, and we can't detect issues at config parsing time unless we add some additional validation logic.

I hope in the future we can get rid of this and resolve `Metastore` and `Storage` with an optional config object:
`Metastore::resolve("s3://...", Some(MetastoreConfig { ... }))`.

File: quickwit-metastore/src/metastore/single_file_metastore/single_file_metastore_factory.rs [hunk 2]
```diff
@@ -40,27 +43,81 @@ impl Default for SingleFileMetastoreFactory {
...
+    static URI_FRAGMENT_PATTERN: OnceCell<Regex> = OnceCell::new();
+    if let Some(captures) = URI_FRAGMENT_PATTERN
+        .get_or_init(|| Regex::new("(.*)#period=([1-9][0-9]{0,8})s").unwrap())
```
Comment (guilload) [inline quickwit-metastore/src/metastore/single_file_metastore/single_file_metastore_factory.rs:49]: `polling_period`?

**Review by guilload — APPROVED** [review]
LGTM.

A few words on naming. `polling_interval` is more commonly used: a search on `polling_interval` vs. `polling_period` on all GitHub yields [837k hits](https://github.com/search?q=polling_interval&type=code) vs. [33k](https://github.com/search?q=polling_period&type=code).

File: quickwit-metastore/src/metastore/single_file_metastore/single_file_metastore_factory.rs (hunk 1 above)
Comment (fulmicoton) [reply quickwit-metastore/src/metastore/single_file_metastore/single_file_metastore_factory.rs:46]: Yes it is not great. Right now if the users gets the DSL wrong we don't spot anything & don't log anything.

File: quickwit-metastore/src/metastore/single_file_metastore/single_file_metastore_factory.rs (hunk 2 above)
Comment (fulmicoton) [reply quickwit-metastore/src/metastore/single_file_metastore/single_file_metastore_factory.rs:49]: renamed polling_interval

**Comment by fulmicoton** [comment]
@guilload Renaming to polling_interval

File: quickwit-metastore/src/metastore/single_file_metastore/single_file_metastore_factory.rs (hunk 2 above)
Comment (fulmicoton) [reply quickwit-metastore/src/metastore/single_file_metastore/single_file_metastore_factory.rs:49]: renaming polling_interval

**Pull request merged**